conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/SocketEnclaveHostInterface.kt

```diff
@@ -86,25 +49,17 @@ class SocketEnclaveHostInterface(
             }
 
             try {
-                /**
-                 * Connect sockets and instantiate data input/output streams.
-                 * We need two sockets here because when running inside a Gramine context, there is some extra
-                 * synchronisation which prevents reading and writing from a socket simultaneously and this can cause
-                 * deadlocks between the receive loop thread and the sending thread. It also improves performance a bit!
-                 * Also set tcpNoDelay to true as latency is more important than throughput in this case.
-                 */
-                fromHostSocket = Socket(host, port).apply { tcpNoDelay = true }
-                toHostSocket = Socket(host, port).apply { tcpNoDelay = true }
-
-                toHost = DataOutputStream(toHostSocket.getOutputStream())
-                fromHost = DataInputStream(fromHostSocket.getInputStream())
-
-                /** Read the maximum number of concurrent calls from the host. */
-                val maximumConcurrentCalls = fromHost.readInt()
-                callExecutor = Executors.newFixedThreadPool(maximumConcurrentCalls)
-
-                /** Start the message receive thread. */
-                receiveLoopThread.start()
+                /** Receive the maximum number of concurrent calls from the host and set up the call executor service. */
+                Socket(host, port).use { initialSocket ->
+                    maximumConcurrentCalls = initialSocket.getInputStream().readInt()
```

Comment:
Can you wrap this inside `DataOutputStream`. Let's try to get rid of the read* methods we ended up creating

Reply:
Done